pkgs/servers/mir/default.nix

```diff
@@ -0,0 +1,121 @@
+{ lib
+, pkgs
+, stdenv
+, fetchFromGitHub
+, cmake
+, extra-cmake-modules
+, boost
+, egl-wayland
+, libglvnd
+, glm
+, glog
+, lttng-ust
+, udev
+, glib
+, wayland
+, libxcb
+, xorg
+, libdrm
+, mesa
+, libepoxy
+, nettle
+, libxkbcommon
+, libinput
+, libxmlxx
+, libuuid
+, pcre2
+, freetype
+, libyamlcpp
+, python3Packages
+, libevdev
+, libselinux
+, libsepol
+, pcre
+, eglexternalplatform
+, doxygen
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mir";
+  version = "2.10.0";
+
+  src = fetchFromGitHub {
+    owner = "MirServer";
+    repo = "mir";
+    rev = "v${version}";
+    sha256 = "sha256-HPbToo7lhsHyx0w4oO2wp3zZ0xpr6SlEm2oE6Rm8b3E=";
+  };
+
+  postPatch = ''
+    # Fix error broken path found in pc file ("//")
+    for f in $(find . -name '*.pc.in') ; do
+      substituteInPlace $f \
+        --replace "/@CMAKE_INSTALL_LIBDIR@" "@CMAKE_INSTALL_LIBDIR@" \
+        --replace "/@CMAKE_INSTALL_INCLUDEDIR@" "@CMAKE_INSTALL_INCLUDEDIR@"
+    done
+
+    # Fix paths for generating drm-formats
+    substituteInPlace src/platform/graphics/CMakeLists.txt \
+      --replace "/usr/include/drm/drm_fourcc.h" "${libdrm.dev}/include/libdrm/drm_fourcc.h" \
+      --replace "/usr/include/libdrm/drm_fourcc.h" "${libdrm.dev}/include/libdrm/drm_fourcc.h"
+
+    # Fix server-platform install path
+    substituteInPlace src/platforms/CMakeLists.txt \
+      --replace "\''${CMAKE_INSTALL_PREFIX}/\''${CMAKE_INSTALL_LIBDIR}/mir/server-platform" "\''${CMAKE_INSTALL_LIBDIR}/mir/server-platform"
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    doxygen
+    extra-cmake-modules
+  ];
+
+  buildInputs = [
+    boost
+    eglexternalplatform
+    egl-wayland
+    freetype
+    glib
+    glm
+    glog
+    libdrm
+    libepoxy
+    libevdev
+    libglvnd
+    libinput
+    libselinux
+    libsepol
+    libuuid
+    libxcb
+    libxkbcommon
+    libxmlxx
+    libyamlcpp
+    lttng-ust
+    mesa
+    nettle
+    pcre
+    pcre2
+    python3Packages.pillow
+    udev
+    wayland
+    xorg.libX11
+    xorg.libXau
+    xorg.libXcursor
+    xorg.libXdmcp
+    xorg.libXrender
+    xorg.xorgproto
+  ];
+
+  cmakeFlags = [
+    "-DMIR_PLATFORM='gbm-kms;eglstream-kms;x11;wayland'"
+    # Tests depend on package wlcs which is not yet packaged
+    "-DMIR_ENABLE_TESTS=OFF"
+  ];
+
+  meta = with lib; {
+    description = "Compositor and display server used by Canonical";
+    homepage = "https://github.com/MirServer/mir";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ onny ];
+  };
```

Comment:
https://discourse.ubuntu.com/t/mir-graphics-support/13185#heading--summary-mir-graphics-platforms
![Bildschirmfoto von 2022-12-27 14-42-40](https://user-images.githubusercontent.com/23431373/209675323-c1675669-99e0-4520-bf9b-8b1ce91009e8.png)

I think all these platforms boil down to Linux-only, more or less?

```suggestion
    platforms = platforms.linux;
  };
```